pkg/archive/archive_test.go

```diff
@@ -1244,6 +1245,8 @@ func TestXGlobalNoParent(t *testing.T) {
 // header entries are created recursively with the default mode (permissions) stored in ImpliedDirectoryMode. This test
 // also verifies that the permissions of explicit directories are respected.
 func TestImpliedDirectoryPermissions(t *testing.T) {
```

Comment:
Rather than skipping this test, can it be run as a non-root user?  I'm assuming it's failing because the [header](https://pkg.go.dev/archive/tar#Header) `Uid` and `Gid` fields are 0 (because they're unset)?  It seems like this test could be run without loss of fidelity if you set `Uid`/`Gid` to the current running user values.

```go
headers := []tar.Header{{
	Name: "deeply/nested/and/implied",
	Uid: os.Getuid(),
	Gid: os.Getgid(),
}}

Reply:
I gave it a go, it seems to have failed for another reason.

```
=== RUN   TestImpliedDirectoryPermissions
    archive_test.go:1272: assertion failed: error is not nil: chown /tmp/TestImpliedDirectoryPermissions1408401907/001/deeply/nested/and: operation not permitted
--- FAIL: TestImpliedDirectoryPermissions (0.00s)
```